internal/bazel_integration_test/test_runner.js

```diff
@@ -0,0 +1,250 @@
+/**
```

Comment:
seems like we should let Ittai and co. know about the additional features we implemented. Not sure if we would want to upstream this or not though

Reply:
I think this is rather different that what they are doing and its also tuned to do package.json replacements which is only relevant for rules_nodejs. But when this lands we should give him a brief maybe there are parts that would useful for him.

Reply:
@gregmagolan, this file (and associated `.bzl` files) are fucking gold. Great work! As the maintainer of the golang verison of this, I'm jealous I do this. I would be interested in helping you upstreaming this if you're open to the idea. Especially if you're interested in trying to do as much as possible in skylark so that it's reusable between all the existing implementations.

This seems like something that would be generally helpful to anyone writing any repo rules, and even more so because it is well abstracted (it's own test rule) and doesn't require you to do anything more than make a workspace file.